controllers/progressivesync_controller.go

```diff
@@ -332,6 +332,28 @@ func (r *ProgressiveSyncReconciler) updateStatusWithRetry(ctx context.Context, p
 	return retryErr
 }
 
+func (r *ProgressiveSyncReconciler) addSyncedAtAnnotation(ctx context.Context, app argov1alpha1.Application, stageName string) error {
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		key := client.ObjectKeyFromObject(&app)
+		latest := argov1alpha1.Application{}
+		if err := r.Client.Get(ctx, key, &latest); err != nil {
+			return err
+		}
+		val, ok := app.Annotations[utils.ProgressiveSyncSyncedAtStageKey]
```

Comment:
Actually, on  a second look, didn't you push the changes to make this `app.ObjectMeta.Annotations`?

Reply:
No, I reverted them as it seems that it cane be done this way as well. I decided to leave it consistent with the rest of the code.

Reply:
Ah - alright, TIL!